python/cudf/cudf/core/join/join.py

```diff
@@ -313,11 +313,10 @@ def _sort_result(self, result: cudf.DataFrame) -> cudf.DataFrame:
             by.extend([result._data[col.name] for col in self._right_keys])
         if by:
             to_sort = cudf.DataFrame._from_data(dict(enumerate(by)))
-            sort_order = ct.as_gather_map(
-                to_sort.argsort(),
+            sort_order = GatherMap.from_column_unchecked(
+                cudf.core.column.as_column(to_sort.argsort()),
```

Comment:
Does this need the same TODO about scattering as above? (Or make the change to use a scatter if that's correct.)

Reply:
This one is a sort-by-key for which there is an open issue, #13557